compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -1432,7 +1432,7 @@ pub fn check_type_bounds<'tcx>(
     let rebased_substs =
         impl_ty_substs.rebase_onto(tcx, impl_ty.container.id(), impl_trait_ref.substs);
 
-    tcx.infer_ctxt().enter(move |infcx| {
+    tcx.infer_ctxt().with_opaque_type_inference(DefiningAnchor::Error).enter(move |infcx| {
```

Comment:
this should not be necessary, the default is to error after all

Reply:
🤦 , fixed it.